compiler/src/dotty/tools/dotc/transform/IsInstanceOfChecker.scala

```diff
@@ -88,19 +88,35 @@ object Checkable {
       }
     }
 
+    def maximize(implicit ctx: Context) = new ApproximatingTypeMap {
+      def apply(tp: Type): Type = tp match {
+        case tp: TypeRef if tp.underlying.isInstanceOf[TypeBounds] =>
+          val lo = this(tp.info.loBound)
+          val hi = this(tp.info.hiBound)
+          if (variance == 0) range(lo, hi)
```

Comment:
The `range` method already does variance checks like this.

Reply:
Thanks 👍